repositories/cyclonedds.BUILD.bazel

```diff
@@ -19,7 +19,7 @@ expand_template(
         "#cmakedefine DDS_HAS_SSL 1": "#undef DDS_HAS_SSL",
         "#cmakedefine DDS_HAS_TYPE_DISCOVERY 1": "#define DDS_HAS_TYPE_DISCOVERY 1",
         "#cmakedefine DDS_HAS_TOPIC_DISCOVERY 1": "#define DDS_HAS_TOPIC_DISCOVERY 1",
-        "#cmakedefine DDS_HAS_SHM 1": "#undef DDS_HAS_SHM",
+        "#cmakedefine DDS_HAS_SHM 1": "#define DDS_HAS_SHM 1",
```

Comment:
I think we should have a `select`statement to conditionally enable this feature for cyclonedds. You can use a bool flag from skylib here; [example](https://github.com/mvukov/qpoases_embedded/blob/master/BUILD.bazel#L4-L14).

Reply:
Whether to enable on shared memory backend is already entirely a[ run time configuration](https://github.com/mvukov/rules_ros2/pull/18/files#diff-6b53520cd82a74f6db2af93ae8e695a697faf73fc5b50f7fc43cec875cfbfc93R5). Cyclone dds is also smart enough to only use shm for publisher/subscriber pairs that fits all the requirements. It will drop back to using network in all cases. Hence I thought it would be ok to always build SHM, it certainly makes the BUILD file simpler and don't need user to do extra steps.